src/sonic_ax_impl/mibs/ietf/rfc4188.py

```diff
@@ -0,0 +1,127 @@
+#import json
+#import ipaddress
+#from enum import unique, Enum
+
+#from swsssdk import port_util
+from sonic_ax_impl import mibs
+from sonic_ax_impl.mibs import Namespace
+from ax_interface import MIBMeta, ValueType, MIBUpdater, SubtreeMIBEntry, MIBEntry
+#from ax_interface.encodings import OctetString
+#from ax_interface.util import mac_decimals, ip2tuple_v4
+from bisect import bisect_right
+#import re
+
+
+class Dot1dBaseTypeConst:
+    unknown = 1
+    transparent = 2
+    source_route = 3
+    srt = 4
+
+class Dot1dBaseUpdater(MIBUpdater):
+    def __init__(self):
+        super().__init__()
+        self.db_conn = Namespace.init_namespace_dbs()
+        self.dot1dbase_port_map = {}
+        self.dot1dbase_port_list = []
+        self.dot1dbase_bridge_addr = None
+        self.dot1d_aging_time = 600
+
+    def reinit_data(self):
+        """
+        Subclass update interface information
+        """
+        Namespace.connect_all_dbs(self.db_conn, mibs.CONFIG_DB)
+        self.dot1dbase_bridge_addr = self.db_conn[0].get(mibs.CONFIG_DB, "DEVICE_METADATA|localhost", 'mac')
+
+    def update_data(self):
+        """
+        Update redis (caches config)
+        Pulls the table references for each vlan member.
+        """
+        self.dot1dbase_port_map = {}
+        self.dot1dbase_port_list = []
+
+        fdb_aging_time = self.db_conn[0].get(mibs.CONFIG_DB, "SWITCH|switch", 'fdb_aging_time')
+        if fdb_aging_time:
+            self.dot1d_aging_time = int(fdb_aging_time)
+        else: 
+            self.dot1d_aging_time = 600 
+
+        vlanmem_entries = Namespace.dbs_keys(self.db_conn, mibs.CONFIG_DB, "VLAN_MEMBER|*")
+        if not vlanmem_entries:
+            return
+
+        for vmem_entry in vlanmem_entries:
+            ifname = vmem_entry.split('|')[2]
+            #if_index = port_util.get_index_from_str(ifname)
+            if_index = mibs.get_index_from_str(ifname)
+            if if_index is None:
+                continue
+            self.dot1dbase_port_map[if_index-1] = if_index
```

Comment:
why is " if_index-1" used as the OID index?
In interface MIB, if index is directly used as OID.
Example: https://github.com/Azure/sonic-snmpagent/blob/master/src/sonic_ax_impl/mibs/ietf/rfc1213.py#L298

Reply:
As per if-mib ifIndex should be (1..2147483647) . if_index is 1 based not zero based.